clang/lib/3C/3C.cpp

```diff
@@ -351,18 +351,25 @@ bool _3CInterface::addVariables() {
 
   std::lock_guard<std::mutex> Lock(InterfaceMutex);
 
+  // first step, so load the ASTs
   ClangTool &Tool = getGlobalClangTool();
+  Tool.buildASTs(ASTs);
```

Comment:
It feels odd to me to build the ASTs here, since the name of the method is `addVariables.` 

Perhaps there should be another method call in `_3CInterface` that you specifically invoke to the build the ASTs, and then the subsequent passes can check to confirm the ASTs have been built already?

Reply:
yep. I'll do a minor refactor once the functionality is stable and I agree this needs its own little section.